elements/pfe-icon/README.md

```diff
@@ -24,77 +24,62 @@ import '@patternfly/pfe-icon';
+### Icon Sets
 
-#### Register a new icon set
+Icon comes with three built-in icon sets:
 
-To register a new icon set, choose a global namespace for that set and identify the path at which the SVGs for that set will be hosted.  Consider also the function needed to convert the icon name into the filename on that hosted location.  The `addIconSet` call accepts the namespace (as a string), the path to the SVGs (as a string), and a function for parsing the icon name into the filename.
+1. `fas`: Font Awesome Free Solid icons (the default set)
+1. `far`: Font Awesome Free Regular icons
+1. `patternfly`: PatternFly icons
 
-```javascript
-await customElements.whenDefined('pfe-icon');
-const PfeIcon = customElements.get('pfe-icon');
-PfeIcon.addIconSet(
-  "local",
-  "./",
-  function(name, iconSetName, iconSetPath) {
-    var regex = new RegExp("^" + iconSetName + "-(.*)");
-    var match = regex.exec(name);
-    return iconSetPath + match[1] + ".svg";
-  }
-);
+Use the `set` attribute to pick an alternative icon set.
+```html
+<pfe-icon icon="star"    set="far"></pfe-icon>
+<pfe-icon icon="redhat"  set="fab"></pfe-icon>
+<pfe-icon icon="package" set="patternfly"></pfe-icon>
 ```
 
-#### Override the default icon sets
-
-Out of the box, the default icon set (using the rh / web namespace) is hosted on [access.redhat.com](https://access.redhat.com). If you would like to override the `rh / web` namespace, you can add the following to a global variable named `PfeConfig`.
-
-The config must be set _before_ the PfeIcon class is defined.
-
-```javascript
-window.PfeConfig = {
-  IconSets: [
-    {
-      name: "web",
-      path: "path/to/svg/directory", // Or https://hosted-icons.com/,
-      resolveIconName: function(name, iconSetName, iconSetPath) { // Optional function to resolve icon paths.
-        var regex = new RegExp("^" + iconSetName + "-(.*)");
-        var match = regex.exec(name);
-        return iconSetPath + match[1] + ".svg";
-      }
-    }
-  ]
-};
-```
+It is possible to add custom icon sets or override the default sets.
+Icon sets are defined in detail in [the docs][icon-sets].
 
-Now when `<pfe-icon>` is used, it will automatically reference the icon set defined in the config.
+### Bundling
 
-If you would like to opt out of any defaults so that you can dynamically add icon sets later using `PfeIcon.addIconSet()`, use the following:
+When bundling PfeIcon with other modules, the default icon imports will be
+relative to the bundle, not the source file, so be sure to either register all
+the icon sets you'll need, or override the default getter.
 
-```javascript
-window.PfeConfig = {
-  IconSets: []
-};
+```js
+// Workaround for bundled pfe-icon: make icon imports absolute, instead of relative to the bundle
+import { PfeIcon } from '/pfe.min.js';
+PfeIcon.getIconUrl = (set, icon) =>
+  new URL(`/assets/icons/${set}/${icon}.js`, import.meta.url);
```

Comment:
There's no chance of race condition here, right?  Importing pfe-icon has a side effect of running `customElements.define("pfe-icon", ...)`, but `PfeIcon.getIconUrl` is guaranteed to run before any  elements start upgrading.  Pretty sure that's the case.

Reply:
I sure:tm: it's good.  `PfeIcon.getIconUrl = ...` will be on the current stack, but any `PfeIcon` constructors run by element upgrades will happen on new stacks further down the queue.  Only if the decorator could somehow cause an element to upgrade immediately would there be a problem and I don't think there's a way to do that.